### PR TITLE
Upgrade pip in travis.

### PR DESCRIPTION
**Pull request by tomprince**
Newer pip will cache wheels for installed packages, which should speed up the build.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -87,10 +87,11 @@ matrix:
 
 # Dependencies installation commands
 install:
+  - pip install -U pip
   # codecov is the interface to codecov.io; see after_success
-  - "echo 'travis_fold:start:codecov'; pip install codecov; echo 'travis_fold:end:codecov'"
-  - "echo 'travis_fold:start:tw'; [ $TWISTED = latest ] || pip install Twisted==$TWISTED; echo 'travis_fold:end:tw'"
-  - "echo 'travis_fold:start:sa'; [ $SQLALCHEMY = latest ] || pip install sqlalchemy==$SQLALCHEMY; echo 'travis_fold:end:sa'"
+  - pip install codecov
```
Comment (tomprince) [inline .travis.yml]: Travis automatically folds individual install steps, so folding again inside the step is a waste (and just makes seeing the contents of the fold more annoying).

**Comment by tomprince** [comment]
Note that because of how cache inheritance works, this will probably only show a benefit after a build of this on master.

**Comment by rutsky** [comment]
Nice side effect: after removing folding we see that there were actually an error: https://travis-ci.org/buildbot/buildbot/jobs/129764461#L334

We need to add `SQLALCHEMY=latest` (or any other version) to `TESTS=trial_worker` specification.

**Comment by tomprince** [comment]
Updated to (hopefully) fix the build.

File: .travis.yml [hunk 2]
```diff
@@ -77,20 +77,22 @@ matrix:
   fast_finish: true
   include:
     # Tests of buildbot-worker and buildbot-slave on python 2.6
+    # Specify SQLALCHEMY=latest to avoid errors installling.
```
Comment (rutsky) [inline .travis.yml]: There is a typo: "insta**lll**ing"

**Pull request merged**